include/dlaf/eigensolver/eigensolver.h

```diff
@@ -40,6 +63,30 @@ EigensolverResult<T, D> eigensolver(blas::Uplo uplo, Matrix<T, D>& mat) {
   return internal::Eigensolver<B, D, T>::call(uplo, mat);
 }
 
+/// Standard Eigensolver.
+///
+/// It solves the standard eigenvalue problem A * x = lambda * x.
+///
+/// On exit, the lower triangle or the upper triangle (depending on @p uplo) of @p mat_a,
+/// including the diagonal, is destroyed.
+///
+/// Implementation on distributed memory.
+///
+/// @param grid is the communicator grid on which the matrix @p mat has been distributed,
+/// @param uplo specifies if upper or lower triangular part of @p mat will be referenced
+/// @param mat contains the Hermitian matrix A
+/// @param eigenvectors matrix of eigenvectors
+/// @param eigenvalues vector of eigenvalues
+template <Backend B, Device D, class T>
+void eigensolver(comm::CommunicatorGrid grid, blas::Uplo uplo, Matrix<T, D>& mat,
+                 Matrix<T, D>& eigenvectors, Matrix<BaseType<T>, D>& eigenvalues) {
+  DLAF_ASSERT(matrix::equal_process_grid(mat, grid), mat);
+  DLAF_ASSERT(square_size(mat), mat);
+  DLAF_ASSERT(square_blocksize(mat), mat);
+
```

Comment:
Could this assert that `eigenvectors` and `eigenvalues` are appropriate sizes? Are there any requirements for the layout and/or distribution?